inspector/src/components/actionTabs/tabs/propertyGrids/animations/addAnimation.tsx

```diff
@@ -26,46 +29,123 @@ export class AddAnimation extends React.Component<
     animationType: string;
     loopMode: number;
     animationTargetPath: string;
+    isUpdating: boolean;
   }
 > {
   constructor(props: IAddAnimationProps) {
     super(props);
+
+    if (this.props.selectedToUpdate !== undefined) {
+      this.setInitialEditingState(this.props.selectedToUpdate);
+    } else {
+      this.setInitialState();
+    }
+  }
+
+  setInitialState() {
     this.state = {
       animationName: '',
       animationTargetPath: '',
       animationType: 'Float',
       loopMode: Animation.ANIMATIONLOOPMODE_CYCLE,
       animationTargetProperty: '',
+      isUpdating: this.props.selectedToUpdate ? true : false,
     };
   }
 
+  setInitialEditingState(editingAnimation: Animation) {

```

Comment:
Can't we unify those two function and have one init/setInitialState function? it feels like the main difference is the (optional?) editingAnimation . Naively:

```javascript
setInitialState(editingAnimation?: Animation) {
this.state = {
      animationName: editingAnimation  ? editingAnimation.name : '',
      animationTargetPath: '',
      animationType: editingAnimation ? this.getTypeAsString(editingAnimation.dataType) : 'Float',
      loopMode: (editingAnimation && editingAnimation.loopMode) ?? Animation.ANIMATIONLOOPMODE_CYCLE,
      animationTargetProperty: editingAnimation ? editingAnimation.targetProperty : '',
      isUpdating: editingAnimation ? true : (this.props.selectedToUpdate ? true : false),
    };
}
```

Of course, needs to be checked :-)

Reply:
That is a great option. Thanks. I was using two functions to track when the component is in edit mode and add new mode...